process/interceptors/factory/interceptedEquivalentProofsFactory.go

```diff
@@ -0,0 +1,44 @@
+package factory
+
+import (
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/dataRetriever"
+	"github.com/multiversx/mx-chain-go/process"
+	"github.com/multiversx/mx-chain-go/process/block/interceptedBlocks"
+	"github.com/multiversx/mx-chain-go/sharding"
+)
+
+type interceptedEquivalentProofsFactory struct {
+	marshaller        marshal.Marshalizer
+	shardCoordinator  sharding.Coordinator
+	headerSigVerifier consensus.HeaderSigVerifier
+	headers           dataRetriever.HeadersPool
+}
+
+// NewInterceptedEquivalentProofsFactory creates a new instance of interceptedEquivalentProofsFactory
+func NewInterceptedEquivalentProofsFactory(args ArgInterceptedDataFactory) *interceptedEquivalentProofsFactory {
+	return &interceptedEquivalentProofsFactory{
+		marshaller:        args.CoreComponents.InternalMarshalizer(),
+		shardCoordinator:  args.ShardCoordinator,
+		headerSigVerifier: args.HeaderSigVerifier,
+		headers:           args.HeadersPool,
```

Comment:
why do we need the headers pool?
With the latest version of the proof we do not need to check any header, just the proof itself.